crates/turbopack-node/src/pool.rs

```diff
@@ -363,11 +371,17 @@ impl NodeJsOperation {
 
 impl Drop for NodeJsOperation {
     fn drop(&mut self) {
-        if let Some(process) = self.process.take() {
-            self.processes
-                .lock()
-                .unwrap()
-                .push(NodeJsPoolProcess::Running(process));
+        if let Some(mut process) = self.process.take() {
+            if self.in_bad_state {
+                if let Some(mut child) = process.child.take() {
+                    let _ = child.start_kill();
+                }
+            } else {
+                self.processes
+                    .lock()
+                    .unwrap()
+                    .push(NodeJsPoolProcess::Running(process));
+            }
         }
```

Comment:
```suggestion
            if !self.in_bad_state {
                self.processes
                    .lock()
                    .unwrap()
                    .push(NodeJsPoolProcess::Running(process));
            }
        }
```

`self.process` will already kill the child

Reply:
actually, we can remove the other `child.kill` drop impls above and add `cmd.kill_on_drop(true);` to `NodeJsPoolProcess::new`

Reply:
I'm fine with the loop for now, but I'd like this changed